upstairs/src/test.rs

```diff
@@ -3668,20 +3668,206 @@ mod test {
+        let max_gen = 33;
+        ds.convert_rc_to_messages(rec_list, max_flush, max_gen);
+
+        // Walk the list and check for messages we expect to find
+        assert_eq!(ds.reconcile_task_list.len(), 4);
+
+        // First task, flush
+        let rio = ds.reconcile_task_list.pop_front().unwrap();
+        assert_eq!(rio.id, 0);
+        match rio.op {
+            Message::ExtentFlush(rep_id, ext, source, mf, mg) => {
+                assert_eq!(rep_id, 0);
+                assert_eq!(ext, repair_extent);
+                assert_eq!(source, 0);
+                assert_eq!(mf, max_flush);
+                assert_eq!(mg, max_gen);
+            }
+            m => {
+                panic!("{:?} not ExtentFlush()", m);
+            }
+        }
+        assert_eq!(Some(&IOState::New), rio.state.get(&0));
+        assert_eq!(Some(&IOState::New), rio.state.get(&1));
+        assert_eq!(Some(&IOState::New), rio.state.get(&2));
+
+        // Second task, close extent
+        let rio = ds.reconcile_task_list.pop_front().unwrap();
+        assert_eq!(rio.id, 1);
+        match rio.op {
+            Message::ExtentClose(rep_id, ext) => {
+                assert_eq!(rep_id, 1);
+                assert_eq!(ext, repair_extent);
+            }
+            m => {
+                panic!("{:?} not ExtentClose()", m);
+            }
+        }
+        assert_eq!(Some(&IOState::New), rio.state.get(&0));
+        assert_eq!(Some(&IOState::New), rio.state.get(&1));
+        assert_eq!(Some(&IOState::New), rio.state.get(&2));
+
+        // Third task, repair extent
+        let rio = ds.reconcile_task_list.pop_front().unwrap();
+        assert_eq!(rio.id, 2);
+        match rio.op {
+            Message::ExtentRepair(rep_id, ext, source, repair, dest) => {
+                assert_eq!(rep_id, rio.id);
+                assert_eq!(ext, repair_extent);
+                assert_eq!(source, 0);
+                assert_eq!(repair, r0);
+                assert_eq!(dest, vec![1, 2]);
+            }
+            m => {
+                panic!("{:?} not ExtentRepair", m);
+            }
+        }
+        assert_eq!(Some(&IOState::New), rio.state.get(&0));
+        assert_eq!(Some(&IOState::New), rio.state.get(&1));
+        assert_eq!(Some(&IOState::New), rio.state.get(&2));
+
+        // Third task, close extent
+        let rio = ds.reconcile_task_list.pop_front().unwrap();
+        assert_eq!(rio.id, 3);
+        match rio.op {
+            Message::ExtentReopen(rep_id, ext) => {
+                assert_eq!(rep_id, 3);
+                assert_eq!(ext, repair_extent);
+            }
+            m => {
+                panic!("{:?} not ExtentClose()", m);
+            }
+        }
+        assert_eq!(Some(&IOState::New), rio.state.get(&0));
+        assert_eq!(Some(&IOState::New), rio.state.get(&1));
+        assert_eq!(Some(&IOState::New), rio.state.get(&2));
+    }
+    /*
+     *
+              let result = std::panic::catch_unwind(||
```

Comment:
commented out code :)

Reply:
Removed.

That was my trying to figure out how to update some of the  #should_panic tests to only accept panic
on the exact place, and not anywhere in the test (like they do now).  I could not get it working so I
punted on fixing that for now, but forgot to remove my sample code.  It will be remove.

Reply:
Gone